src/core/Directus/Application/CoreServicesProvider.php

```diff
@@ -694,7 +695,7 @@ protected function getEmitter()
                 $acl = $container->get('acl');
                 $tableGateway = new BaseTableGateway(SchemaManager::COLLECTION_ROLES, $zendDb, $acl);
                 $row = $tableGateway->select(['id' => $roleId])->current();
-                if (strtolower($row->name) === 'public') {
+                if (strtolower($row->name) === Roles::PUBLIC) {
```

Comment:
Will this work? because before it was checking that `'public' === 'public'`, now it's checking that `'public' === 2`